### ISSUE TITLE
Parameterized Decorators don't receive value change events

### ISSUE THREAD

**Issue by marcello3d** [ractivejs/ractive#367]
I've been experimenting with decorators in attempt to insert custom widgets into my ractive templates.

I created the decorator with:

``` js
Ractive.decorators.widget = function(node, options) {
    console.log("create widget", this, node, options)
    return {
        teardown: function() {
            console.log("teardown", this, arguments)
        }
    }
}
```

And then created them in a loop:

``` html
{{#widgets}}
<div decorator="widget:{{this}}"></div>
{{/widgets}}
```

The issue is that when I change the widgets list, I only get creates or teardowns when there's a change in the length of the list, never if the contents of the widgets change.

I'm guessing this is because you're reusing the dom nodes, so I'd either like to see the teardown/create happen regardless, or better yet, be able to bind to changes in the options passed to my widget. Ideally it'd also be nice to know when some subtree of the decorated node changes.

In short I'm trying to make custom widgets that aren't rendered by Ractive, but show up inside my Ractive layout.

Any workaround you can think of?

Thank you!
